src/main/java/org/ow2/proactive/procci/rest/ComputeRest.java

```diff
@@ -121,14 +102,12 @@
     public ResponseEntity<ResourceRendering> getCompute(@PathVariable("id") String id) {
         logger.debug("Get Compute ");
         try {
-            Optional<Model> computeModel = instancesServices.getInstanceByVariable(ID_NAME,
-                    ConvertUtils.formatURL(id));
-            if (!computeModel.isPresent()) {
+
+            Optional<Entity> compute = instancesService.getEntity(ConvertUtils.formatURL(id));
```

Comment:
If you keep the existing signature, you could most probably do a one-line with _compute.map(...).orElse(...);_